lib/maths/CGammaRateConjugate.cc

```diff
@@ -658,11 +657,15 @@ class CLogMarginalLikelihood : core::CNonCopyable {
 
             LOG_TRACE(<< "numberSamples = " << m_NumberSamples);
 
-            m_Constant = m_PriorShape * std::log(m_PriorRate) -
-                         boost::math::lgamma(m_PriorShape) +
+            m_Constant = m_PriorShape * std::log(m_PriorRate) - std::lgamma(m_PriorShape) +
                          logVarianceScaleSum - logGammaScaledLikelihoodShape -
-                         nResidual * boost::math::lgamma(m_LikelihoodShape) +
-                         boost::math::lgamma(m_ImpliedShape);
+                         nResidual * std::lgamma(m_LikelihoodShape) +
+                         std::lgamma(m_ImpliedShape);
+
+            if (std::isfinite(m_ImpliedShape) && std::isfinite(m_Constant) == false) {
```

Comment:
Surely the binding is wrong for this: don't you want this condition to be `(std::isfinite(m_ImpliedShape) && std::isfinite(m_Constant)) == false`.

Reply:
good catch